controllers/barbican_common.go

```diff
@@ -52,12 +57,94 @@ func GenerateConfigsGeneric(
 	}
 	if scripts {
 		cms = append(cms, util.Template{
-			Name:         fmt.Sprintf("%s-scripts", instance.GetName()),
-			Namespace:    instance.GetNamespace(),
-			Type:         util.TemplateTypeScripts,
-			InstanceType: instance.GetObjectKind().GroupVersionKind().Kind,
-			Labels:       cmLabels,
+			Name:          fmt.Sprintf("%s-scripts", instance.GetName()),
+			Namespace:     instance.GetNamespace(),
+			Type:          util.TemplateTypeScripts,
+			InstanceType:  instance.GetObjectKind().GroupVersionKind().Kind,
+			ConfigOptions: templateParameters,
+			Labels:        cmLabels,
 		})
 	}
 	return secret.EnsureSecrets(ctx, h, instance, cms, envVars)
 }
+
+func GenerateSecretStoreTemplateMap(
+	enabledSecretStores []barbicanv1beta1.SecretStore,
+	globalDefaultSecretStore barbicanv1beta1.SecretStore,
+) (map[string]interface{}, error) {
+	// Log := r.GetLogger(ctx)
+	stores := []string{}
+	if len(enabledSecretStores) == 0 {
+		stores = []string{"simple_crypto"}
+	} else {
+		for _, value := range enabledSecretStores {
+			stores = append(stores, string(value))
+		}
+	}
+
+	if len(globalDefaultSecretStore) == 0 {
+		globalDefaultSecretStore = "simple_crypto"
+	}
+
+	tempMap := map[string]interface{}{
+		"EnabledSecretStores":      strings.Join(stores, ","),
+		"GlobalDefaultSecretStore": globalDefaultSecretStore,
+		"SimpleCryptoEnabled":      slices.Contains(stores, "simple_crypto"),
+		"PKCS11CryptoEnabled":      slices.Contains(stores, "pkcs11"),
+	}
+	return tempMap, nil
+}
+
+func GeneratePKCS11TemplateMap(
+	ctx context.Context,
+	h *helper.Helper,
+	pkcs11 barbicanv1beta1.BarbicanPKCS11Template,
+	namespace string,
+) (map[string]interface{}, error) {
+	tempMap := map[string]interface{}{}
+	hsmLoginSecret, _, err := oko_secret.GetSecret(ctx, h, pkcs11.HSMLoginSecret, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Checking if at least one of the following parameters has been provided.
+	if len(pkcs11.HSMTokenSerialNumber) == 0 && len(pkcs11.HSMTokenLabels) == 0 && len(pkcs11.HSMSlotId) == 0 {
```

Comment:
Investigate whether we could use webhooks for validation

Reply:
ack - lets leave this to a follow-on patch.  I started to look at this, but its a little more involved than I initially thought.

Reply:
done